File: ProductPage.java
package haripriyapackage;
import org.openqa.selenium.*;
import org.openqa.selenium.chrome.chromedriver;
import org.testng.Assert;
import org.testng.annotations.*;

public class ProductPage {
    private WebDriver driver;

    private By addToCartButton = By.xpath("//button[text()='Add to cart']");

    public ProductPage(WebDriver driver) {
        this.driver = driver;
    }

    public CartPage addToCart() {
        driver.findElement(addToCartButton).click();
		Set<String> handles=driver.getWindowHandles();
		driver.switchTo().window().last();
        return new CartPage(driver);
    }
}

File: pythonproject/product_page.py
from base_page import BasePage

class ProductPage(BasePage):
    ADD_TO_CART_BUTTON = (By.XPATH, "//button[text()='Add to cart']")

    def add_to_cart(self):
        self.driver.find_element(*self.ADD_TO_CART_BUTTON).click()
        self.switch_to_new_window()
        return CartPage(self.driver)  # Return CartPage instance
